### PR TITLE
Initial phase of the Role management feature is implemented to the B2B sample application

### PR DESCRIPTION
**Pull request by Achintha444**
## Purpose
Initial phase of the Role management feature is implemented to the B2B sample application. 
- Now a user can view roles and edit the role name from the sample application.

Role management interface
Screenshot 2022-11-11 at 15 46 34

### CODE REVIEW

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleButton.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * 
+ * @returns - create role button
+ */
+export default function CreateRoleButton() {
+
+    return (
+        <Button
+            appearance="primary"
+            size="lg"
+            onClick={ ()=>{} }>
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleButton.js:34]: Why is this an empty function?

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    };
+
+    const roleItemDetailsComponent = (activeKey) => {
+        switch (activeKey) {
+            case "1":
+
+                return <General session={ session } roleDetails={ roleDetails } fetchData={ fetchData } />;
+                // case "2":
+
+            //     return <Settings session={session} idpDetails={idpDetails} />;
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js:61]: Why has this been commented out?

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+/**
+ * 
+ * @param {*} session - session
+ * @param {*} roleUri - uri of the role
+ * @param {*} patchMethod - `PatchMethod` value
+ * @param {*} path - path variable (`users` | `permission` | `displayName`)
+ * @param {*} value - value that need to be updated. If `patchMethod` is `PatchMethod.ADD` then `value` 
+ * should be an `array`
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js:69]: Shal we remove these asterisks?

**Review by thivi — APPROVED** [review]

**Review by thivi — APPROVED** [review]

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleButton.js (hunk 1 above)
Comment (lashinijay) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleButton.js]: was this intended?

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js (hunk 2 above)
Comment (lashinijay) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js]: please provide a reason.

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            <div
+                style={ {
+                    alignItems: "stretch",
+                    display: "flex"
+                } }>
+                <Nav.Item
+                    eventKey="1"
+                    onSelect={ (eventKey) => activeKeyNavSelect(eventKey) }>General</Nav.Item>
+
+                { /* <Nav.Item
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js:105]: Should we keep the commented lines? Can't we remove them if thy are not needed?

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        if (res) {
+            setRolesList(res);
+        } else {
+            setRolesList([]);
+        }
+
+    }, [ session ]);
+
+    // const onAddIdentityProviderClick = () => {
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js:55]: why these are commented?

File: b2b-sample/guardio-insurance-sample-application/components/common/emptySettings.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+ */
+export default function EmptySettings(prop) {
+
+    const { bodyString, buttonString, icon, onAddButtonClick } = prop;
+
+    return (
+        <FlexboxGrid
+            style={ { height: "60vh", marginTop: "24px", width: "100%" } }
+            justify="center"
+            align="middle">
```
[the remark this answers is not in the document]
Comment (lashinijay) [reply b2b-sample/guardio-insurance-sample-application/components/common/emptySettings.js]: Please address these comments.

File: b2b-sample/guardio-insurance-sample-application/util/util/common/common.js [hunk 7]
```diff
@@ -73,6 +73,24 @@ function copyTheTextToClipboard(text, toaster) {
...
+ * 
+ * @returns random generatored rgb colour
+ */
+function random_rgba() {
+    var o = Math.round, r = Math.random, s = 255;
+    return 'rgba(' + o(r()*s) + ',' + o(r()*s) + ',' + o(r()*s) + ',' + r().toFixed(1) + ')';
+}
+
+/**
+ * operations that we can do on PATCH methods
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/util/util/common/common.js:86]: `Operations that we can do on PATCH methods.` Formatting issues check other places too.

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+            return getAddReplaceBody(patchMethod, path, value);
+        case PatchMethod.REMOVE:
+
+            return getRemoveBody(patchMethod, path, value);
+    }
+}
+
+/**
+ *
```
Comment (ThaminduR) [inline b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js:63]: Shall we add function descriptions. Check other places as well.

File: b2b-sample/guardio-insurance-sample-application/components/common/emptySettings.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/common/emptySettings.js]: Fixed with https://github.com/Achintha444/samples-is/commit/728f64d702c2d65ab22f2c0cab5cc3a4ae6b76c0

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js (hunk 2 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js]: This is already fixed from the commit https://github.com/Achintha444/samples-is/commit/0db49b892ae09126a0bb153033074fa0a06ec2ba#diff-48d3a4aa1e8e5111ed172799200ba92ea05adc72937247d83c550ee7ce702e69

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleButton.js (hunk 1 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleButton.js]: This was fixed when the functionality of this file moved to a different location https://github.com/Achintha444/samples-is/commit/28b207622f0bf667fffe757611fbb7a8983a8429#diff-ae1fe67e81a5d047523b231d7112e52f5c53467b16d8500212e19f7fe856e144

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js (hunk 3 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js]: Fixed when the functionality moved to a different location 
https://github.com/Achintha444/samples-is/commit/b8a143e14872c6b7a91d0f029bfffffa5fea66ef#diff-e70752c7d2a32606406568ac366fe30cf2f7c3ce42ec873152e4f583c38e394f

File: b2b-sample/guardio-insurance-sample-application/util/util/common/common.js (hunk 7 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/util/common/common.js:86]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/8b059f39358c5652ea99e473f4cf34e23ad95e0b

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js (hunk 4 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js:105]: This is already fixed when the functionality moved to a different location
https://github.com/Achintha444/samples-is/commit/537311db23374d1deff616fdcd3f1d93b7309cbe#diff-bb384cb53b82d87ca1a8aabb3f7b399758ebe30958a67a8e04fbabe56ede519a

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js (hunk 5 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js:55]: Fixed when the functionality moved to a different location 
https://github.com/Achintha444/samples-is/commit/8288cbd899e674b70a91604fa51d6d65a520c2b2#diff-fc895b6281bfab3b91148d2efca61917ce6be14868c2de023069cf5438f42c7b

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js (hunk 8 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/role/decodePatchRole.js:63]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/732b9bcc7aa8badcd42dd91959331576e9a105ab

**Pull request merged**